### PR TITLE
Updates to getting started, GLM and model comparison notebooks

### PR DESCRIPTION
**Pull request by fonnesbeck**
This updates the GLM and model comparison notebooks to use v4.0 beta.

### CODE REVIEW

**Comment by tomicapretto** [comment]
<p>You're right. Bambi still relies on v3 and it uses Theano.</p><p> </p><p>The development version is not able to run on v4, but I think it is something we could implement without much difficulty. There's only one thing that may be problematic and it is related to the <code>LKJCholeskyCov</code>   distribution. I don't know if it is already working on v4 already.</p><p> </p><p>I plan to make a new release in the first half of January. We could work on porting Bambi development version to v4 after that.&nbsp;</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>I had theano-pymc installed in the env, which caused this. Goes away when I remove it.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Review by OriolAbril — CHANGES_REQUESTED** [review]
marking with request changes to make sure we update the text to reflect what the code does and the current situation.

I also think it should not use first person "I" anymore as it's no longer a blogpost but a collaborative page in the docs.

**Comment by fonnesbeck** [comment]
Draft of getting started notebook added. Will finish it up in the next few hours.

**Comment by OriolAbril** [comment]
could we rename that notebook to `intro_for_ppl_practitioners` or something along these lines? 

I think it is a great notebook _for this audience_ (which imo is what it was aiming for, after all this is based on the paper whose audience is clearly ppl practitioners who keep up with the literature), but it's not really a getting started notebook. New users writing their first bayesian model with a ppl should not need arbitrary logprob terms, distributions nor using aesara explicitly and therefore none of that should be mentioned when writing a [tutorial](https://diataxis.fr/tutorials/). I think having a "pressing void" on a getting started for pymc newcomers might make it a bit more of a priority. 

On a related note, maybe we could base the getting started notebook on @AustinRochford webinar for the upcoming pymc-data umbrella series which could then be notebook with a talk.

**Comment by twiecki** [comment]
Completely agree @OriolAbril. I think we want a "Getting started if you know Bayesian modeling" and one if you don't.

**Comment by fonnesbeck** [comment]
I'm all for that, but we should probably not let that be a blocker for getting out of beta. 

How about we rename this one `pymc_overview.ipynb` which is a more accurate description? We can then add the intro for non-Bayesians notebook later. Its not clear whether that would belong here or with pymc-examples.

**Comment by OriolAbril** [comment]
> I'm all for that, but we should probably not let that be a blocker for getting out of beta.

agreed, the blockers should be having everything in the api following the template in #5266 and addressing the TODOs currently visible in the website. @martinacantaro knows better and we'll try to organize when she is back after new year.

> How about we rename this one pymc_overview.ipynb which is a more accurate description? 

that is perfect, we do need to update the toctrees now though so the notebook appears in the website, relevant sidebars and so on, not sure if I'll have time to do this sometime soon.

> We can then add the intro for non-Bayesians notebook later. Its not clear whether that would belong here or with pymc-examples.

It should definitely be here I think. The idea was to have a handful of notebooks included in the versioned docs (and therefore always executed with the version of the docs, I'll try to set up CI for this) covering the core functionalities of pymc.

**Comment by twiecki** [comment]
<p>I'd either update the cell or just manually type out that this was on version 4.0.0b1. We update these NB so infrequently that we can expect these warnings to be here for a long time.</p><p> </p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by OriolAbril** [comment]
<p>going forward, this might still be true for pymc-examples but won't and can't be the case with these notebooks. Moving them back here and doing so with only a handful is to make them part of the versioned docs. Will try to set up CI to autoexecute and everything, but these notebooks need to always be executed with the current version of the documentation. If that can't be for some reason (i.e. too long to run on ci) then they have to be moved back to pymc-examples.</p><p> </p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>Removed</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>I've revised all this now.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>Fixed.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>Fixed.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>Should be fixed now. bambi installs theano-pymc as a dependency, which is not ideal, but I assume that goes away in the near future when it is updated to PyMC4/Aesara.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>Fixed</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>Fixed </p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>Removed the initial value altogether</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: README.rst [hunk 1]
```diff
@@ -10,7 +10,7 @@ focusing on advanced Markov chain Monte Carlo (MCMC) and variational inference (
 algorithms. Its flexibility and extensibility make it applicable to a
 large suite of problems.
 
-Check out the `getting started guide <https://docs.pymc.io/en/stable/pymc-examples/examples/getting_started.html>`__,  or
+Check out the `PyMC overview <https://docs.pymc.io/en/stable/pymc-examples/examples/pymc_overview.html>`__,  or
```
Comment (OriolAbril) [inline README.rst]:
```suggestion
Check out the `PyMC overview <https://docs.pymc.io/en/stable/learn/examples/pymc_overview.html>`__,  or
```

File: README.rst [hunk 2]
```diff
@@ -41,7 +41,7 @@ If you already know about Bayesian statistics:
 ----------------------------------------------
 
 -  `API quickstart guide <https://docs.pymc.io/en/stable/pymc-examples/examples/pymc3_howto/api_quickstart.html>`__
--  The `PyMC tutorial <https://docs.pymc.io/en/stable/pymc-examples/examples/getting_started.html>`__
+-  The `PyMC tutorial <https://docs.pymc.io/en/stable/pymc-examples/examples/pymc_overview.html>`__
```
Comment (OriolAbril) [inline README.rst]:
```suggestion
-  The `PyMC tutorial <https://docs.pymc.io/en/stable/learn/examples/pymc_overview.html>`__
```

**Review by OriolAbril — COMMENTED** [review]
these links won't work yet I think because the name change will have an effect on latest only, not stable. but once v4 is released this is where the notebook will be (and stay)

**Comment by twiecki** [comment]
<p>These do not look that great, or is that the point? Why not pick non-fat-tailed priors?</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by twiecki** [comment]
<p>Still see it: switchpoint = pm.DiscreteUniform(</p><p>&nbsp;&nbsp;&nbsp;&nbsp;"switchpoint", lower=years.min(), upper=years.max(), testval=1900</p><p>&nbsp;&nbsp;)</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by twiecki** [comment]
I think these could be great getting started guides already. one for if you do not know Bayesian modeling, one if you do.

**Comment by fonnesbeck** [comment]
<p>Its a horseshoe prior. If we chose normal priors we would not be doing sparse regression.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>I can remove it.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>I think its important to show that other samplers can be used. There's nothing wrong with the slice sampler. </p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>Good idea.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>No, that strips away all the x-axes labels except for the bottom one.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>The plot looks better but it does a poor job with the labels (see newly-pushed version)</p><p> </p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>I can't seem to get any of these working here. I will leave them as markdown for now.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by OriolAbril** [comment]
<p>Oh yeah, it looks like ArviZ sets the grid and ticklables to match the centers of the bins. Changing the order should look better but we should probably add a way to turn this off in ArviZ and have regular grid and ticklabels not tied to any histogram (which doesn't make much sense when comparing)</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by OriolAbril** [comment]
<p>wasn't sure it'd work, let's forget about this</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>The problem here is that I want the text to say "platform-specific installation guides" and not simply "installation"</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>Similar to above, I don't want the text to say "api" I want it to say "API documentation".</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by OriolAbril** [comment]
There are some examples of custom text in https://sphinx-primer.readthedocs.io/en/latest/core/roles.html

**Comment by fonnesbeck** [comment]
<p>I've updated these passages</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>done</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
<p>Unfortunately swapping the order creates xticklabels that overlap to the point of unreadability. I have left it as is for the time being.</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5269/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by fonnesbeck** [comment]
I think I've addressed all of the comments as best as possible. Let me know if there is anything else that needs attention.

**Comment by OriolAbril** [comment]
```
{ref}`API documentation <api>`
{ref}`platform-specific installation guides <installation>`
```

Will point to the right place and show the custom text. There shouldn't be links hardcoded to v3 docs unless they aim to always point there.

**Review by OriolAbril — APPROVED** [review]

**Pull request merged**